File: lib-content-scrapers/src/main/java/com/ustadmobile/lib/contentscrapers/khanacademy/ModuleResponse.java
package com.ustadmobile.lib.contentscrapers.khanacademy;

import com.ustadmobile.lib.contentscrapers.util.DownloadUrl;

import java.util.List;

public class ModuleResponse {

    public String description;

    public String icon;

    public String kind;

    public String slug;

    public String title;

    public String topicId;

    public String url;

    public List<Tutorial> tutorials;

    public List<ModuleResponse> modules;

    public class Tutorial {

        public List<ContentItem> contentItems;

        public String description;

        public String id;

        public String slug;

        public String title;

        public String url;

        public class ContentItem {

            public String contentId;

            public String description;

            public String kind;

            public String thumbnailUrl;

            public String slug;

            public String nodeUrl;

            public String title;

            public int expectedDoNCount;

            public DownloadUrl downloadUrls;

        }
    }

    public SubjectChallenge subjectChallenge;

    public class SubjectChallenge {

        // TODO

    }

}

File: lib-content-scrapers/src/jvmMain/kotlin/com/ustadmobile/lib/contentscrapers/khanacademy/ModuleResponse.kt
package com.ustadmobile.lib.contentscrapers.khanacademy

import com.ustadmobile.lib.contentscrapers.util.DownloadUrl

class ModuleResponse {

    var description: String? = null

    var icon: String? = null

    var kind: String? = null

    var slug: String? = null

    var title: String? = null

    var topicId: String? = null

    var imageUrl: String? = null

    var url: String? = null

    var tutorials: List<Tutorial>? = null

    var modules: List<ModuleResponse>? = null

    var contentItems: List<Tutorial.ContentItem>? = null

    var subjectChallenge: SubjectChallenge? = null

    inner class Tutorial {

        var contentItems: List<ContentItem>? = null

        var description: String? = null

        var id: String? = null

        var slug: String? = null

        var title: String? = null

        var url: String? = null

        inner class ContentItem {

            var contentId: String? = null

            var description: String? = null

            var kind: String? = null

            var thumbnailUrl: String? = null

            var slug: String? = null

            var nodeUrl: String? = null

            var title: String? = null

            var expectedDoNCount: Int = 0

            var downloadUrls: DownloadUrl? = null

        }
    }

    inner class SubjectChallenge// TODO

}
